src/SignalR/server/SignalR/test/HubConnectionHandlerTests.cs

```diff
@@ -2911,6 +2911,37 @@ bool ExpectedErrors(WriteContext writeContext)
             }
         }
 
+        [Fact]
+        public async Task StreamUploadBufferCapacityBlocksOtherInvocations()
+        {
+            var serviceProvider = HubConnectionHandlerTestUtils.CreateServiceProvider(services =>
+            {
+                services.Configure<HubOptions>(options =>
+                {
+                    options.StreamBufferCapacity = 1;
+                });
+            });
+
+            var connectionHandler = serviceProvider.GetService<HubConnectionHandler<MethodHub>>();
+
+            using (var client = new TestClient())
+            {
+                var connectionHandlerTask = await client.ConnectAsync(connectionHandler).OrTimeout();
+                await client.BeginUploadStreamAsync("invocation", nameof(MethodHub.StreamDontRead), new[] { "id" }, Array.Empty<object>());
+
+                foreach (var letter in new[] { "A", "B", "C", "D", "E" })
+                {
+                    await client.SendHubMessageAsync(new StreamItemMessage("id", letter)).OrTimeout();
+                }
+
+                var ex = await Assert.ThrowsAsync<TimeoutException>(async () =>
+                {
+                    await client.SendInvocationAsync("Echo", "test");
+                    var result = (CompletionMessage)await client.ReadAsync().OrTimeout(5000);
```

Comment:
:( not a fan

Reply:
Not a huge fan either but it does demonstrate the behavior we want.

Reply:
Can we verify that the CompletionMessage eventually arrives after the hub starts reading the stream items?

Reply:
Talked about this irl with Stephen. This gets messy because you have to synchronize blocking reading the incoming stream items to verify processing invocations is blocked, then you'd have to unblock (the actual tricky part) to verify that invocations can continue to be processed.